pages/join/[...joinCode].js

```diff
@@ -2,110 +2,110 @@ import Head from 'next/head';
 import Navbar from '../../components/navbar';
 import { useState } from 'react';
 import { useRouter } from 'next/router';
+import { getSession } from 'next-auth/react';
+import AuthButton from '../../components/authButton';
 
-export default function JoinWithCode() {
-  const [formData, setFormData] = useState({});
-
-  async function saveEmail(e) {
-    e.preventDefault();
-    const response = await fetch(`/api/student_email_join`, {
-      method: 'PUT',
-      body: JSON.stringify(formData)
-    });
-    const status = response.status;
-    if (status === 400) {
-      alert('This email is not associated with freeCodeCamp');
-    } else if (status === 409) {
-      alert('This account is already signed up');
-    } else {
-      alert('Successfully registered, please exit window');
+export async function getServerSideProps(ctx) {
+  const userSession = await getSession(ctx);
+  return {
+    props: {
+      userSession: userSession
     }
-  }
-
+  };
+}
+export default function JoinWithCode({ userSession }) {
+  const [formData] = useState({});
   const router = useRouter();
   const { joinCode } = router.query;
+
+  const classroomRequest = async event => {
+    event.preventDefault();
+    formData.join = joinCode;
+    try {
+      const res = await fetch(`/api/student_email_join`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(formData)
+      });
+      if (res.status === 409) {
+        alert('You have already joined this classroom.');
```

Comment:
It would be better to generate the error message on the server (in api/student_email_join) and then have the browser display any errors that the server sends over.  For now, it's ok to leave it as is, but I think we should file an issue to refactor the error message handling eventually.